kgforge/specializations/models/rdf/service.py

```diff
@@ -187,23 +220,49 @@ def generate_context(self) -> Dict:
         raise NotImplementedError()
 
     @abstractmethod
-    def _build_shapes_map(self) -> Dict:
-        """Queries the source and returns a map of owl:Class to sh:NodeShape"""
+    def _build_shapes_map(self) -> Tuple[Dict, Dict, Dict]:
+        """Index the loaded node shapes
+        Returns:
+            * a Dict of a targeted owl:Class to a shape
+            * a Dict of a shape to the resource defining it through <https://bluebrain.github.io/nexus/vocabulary/shapes>
+            * a Dict of a resource defining a shape to the a named graph containing it
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def load_shape_graph(self, graph_id: str, schema_id: str) -> Graph:
```

Comment:
I think this should have support for graph_id and schema_id being either str of URIRef, and that all code that does self._graph.graph(...) should use this method instead

Reply:
4 calls actually:

1. 1 in service._transitive_load_shape_graph
2. 1 in service.get_shape_graph
3. 1 in directory_service.load_shape_graph
4. 1 in store_service.load_shape_graph

service.get_shape_graph is the method to call to get a shape graph from a shape URI (not load_shape_graph).
load_shape_graph is for loading a shape in service._graph from  source (dir or store). To clarify, I renamed it to load_shape_graph_from_source. 

service.get_shape_graph  can't be used in 1, 3 and 4 because service.get_shape_graph actually uses them. These services are internal to the RDFModel class, when these features to get shape graph will be added to RDFModel then we can support Union of type

Reply:
I meant that there are multiple calls to self._dataset_graph.graph(...) that can be replaced with load_shape_graph_from_source, they all start from a graph id. 
In some cases it's a str, in others it's already a URIRef, so it would be best to call load_shape_graph_from_source instead of those, and have that function support both str and uriref as input

<img width="708" alt="Screenshot 2024-03-25 at 14 51 13" src="https://github.com/BlueBrain/nexus-forge/assets/24302104/f2a1bc68-8fde-4f85-ac26-1d456791ccca">